packages/babel-core/src/transformation/file/index.js

```diff
@@ -359,21 +359,28 @@ export default class File extends Store {
         sourceRoot: inputMapConsumer.sourceRoot
       });
 
+      // This assumes the output map always has a single source, since Babel always compiles a single source file to a
+      // single output file.
+      const source = outputMapConsumer.sources[0];
+
       inputMapConsumer.eachMapping(function (mapping) {
-        mergedGenerator.addMapping({
-          source: mapping.source,
-
-          original: {
-            line: mapping.originalLine,
-            column: mapping.originalColumn
-          },
-
-          generated: outputMapConsumer.generatedPositionFor({
-            line: mapping.generatedLine,
-            column: mapping.generatedColumn,
-            source: outputMapConsumer.file
-          })
+        const generatedPosition = outputMapConsumer.generatedPositionFor({
+          line: mapping.generatedLine,
+          column: mapping.generatedColumn,
+          source: source
         });
+        if(generatedPosition.column != null) {
```

Comment:
I'm not sure (don't know too much about this) - but your assumption makes sense since the two are tied together.

And source won't be null since we are passing that in from `outputMapConsumer.sources[0];`?